### PR TITLE
initial support for server's settings from intersystems.servers

### PR DESCRIPTION
**Pull request by daimor**

### CODE REVIEW

**Comment by rajrsingh** [comment]
@daimor do you want this approved and merged now?

**Comment by daimor** [comment]
Work in progress, it was happened to be a bit more complicated, than I thought.

File: src/extension.ts [hunk 1]
```diff
@@ -128,46 +135,74 @@ export function getXmlUri(uri: vscode.Uri): vscode.Uri {
...
-    terminalWithDocker();
-    if (dockerPort !== conn.port) {
-      workspaceState.update(currentWorkspaceFolder() + ":port", dockerPort);
+  if (!workspaceState.get(workspaceFolder + ":port") && !api.externalServer) {
+    const { port: dockerPort, docker: withDocker } = portFromDockerCompose();
+    workspaceState.update(workspaceFolder + ":docker", withDocker);
+    if (withDocker) {
+      if (!dockerPort) {
+        outputChannel.appendLine(
+          `Something wrong with your docker-compose connection settings. Or your service not running.`
```
Comment (gjsjohnmurray) [inline src/extension.ts:171]:
```suggestion
          `Something is wrong with your docker-compose connection settings, or your service is not running.`
```

File: src/extension.ts [hunk 2]
```diff
@@ -128,46 +135,74 @@ export function getXmlUri(uri: vscode.Uri): vscode.Uri {
...
   if (clearCookies) {
     api.clearCookies();
+  } else if (connectionSocket && connectionSocket.url == api.xdebugUrl() && connectionSocket.OPEN) {
+    panel.text = `${connInfo} - Connected`;
+    return;
   }
-  if (connectionSocket && connectionSocket.url == api.xdebugUrl() && connectionSocket.OPEN) {
+  api = new AtelierAPI(workspaceFolder);
+  if (!api.config.host || !api.config.port || !api.config.ns) {
+    outputChannel.appendLine("host, port and ns should be specified");
```
Comment (gjsjohnmurray) [inline src/extension.ts:193]:
```suggestion
    outputChannel.appendLine("host, port and ns must be specified.");
```

Comment (gjsjohnmurray) [reply src/extension.ts:193]: Also just noticed that you haven't yet added the dependency on `intersystems-community.servermanager` into `package.json`.

File: package.json [hunk 3]
```diff
@@ -558,29 +558,29 @@
               "description": "Password of username. If not set here it must be provided when connecting."
             },
             "https": {
-              "description": "Use SSL to access to API",
+              "description": "Use SSL to access to API.",
```
Comment (gjsjohnmurray) [inline package.json:561]:
```suggestion
              "description": "Use SSL/TLS to access the server.",
```

**Pull request merged**